test/metabase/public_settings/premium_features_test.clj

```diff
@@ -58,6 +58,7 @@
                                  ;; throw an ex-info here
                                  (throw (Exception. "network issues")))]
           (is (= {:valid         false
+                  :error_code    :unable_to_validate
```

Comment:
It feels a bit weird we need this particular `error_code` to make sure we can differentiate between `invalid_token` and `unable_to_validate` errors.

In FE, we don't even need to check `error_code` we just say anything that's not `unable_to_valiadate`, what if we say if the status is `invalid` then it's `invalid_token` and `unable_to_validate` otherwise? We wouldn't need the BE change would we?

Reply:
You mean this status https://github.com/metabase/metabase/blob/46feb24a931642958c5ce86e84c7a95f33e3b5cb/src/metabase/public_settings/premium_features.clj#L131 ?

Isn't it sent only when the token is an invalid format (ie, not 64 chars)?

When the BE is not able to validate it, it still sends the status but as a translated string: https://github.com/metabase/metabase/blob/46feb24a931642958c5ce86e84c7a95f33e3b5cb/src/metabase/public_settings/premium_features.clj#L155-L156 so we can't really use it in an if.

Reply:
yeah we could say
```ts
// if valid === false
if (status === "invalid")
  errorState = "invalid_token"
else {
  errorState = "unabled_to_validate"
```

Couldn't we? Or did I missed something.

Reply:
Status is "invalid" [only when the token is not formatted correctly](https://github.com/metabase/metabase/blob/46feb24a931642958c5ce86e84c7a95f33e3b5cb/src/metabase/public_settings/premium_features.clj#L129-L132), when the token is of the correct size, but doesn't exist we get a different output.

Here's some sample responses of `premium-features/fetch-token-status`:

invalid 64 chars token:
```json
{
    "valid": false,
    "status": "Token does not exist."
}
``` 

token with wrong format
```json
{
  "valid": false,
  "status": "invalid",
  "error-details": "Token should be 64 hexadecimal characters."
}
```

valid token
```json
{
    "valid": true,
    "status": "Token is valid.",
    "trial": true,
    "valid-thru": ...,
    "features": ...,
    "store-users": ...
}
```
network error (I turned off wifi to simulate)
```json
{
    "valid": false,
    "error_code": "unable_to_validate", // added with this pr
    "status": "Unable to validate token",
    "error-details": "token-check.metabase.com: nodename nor servname provided, or not known"
}
```

The issue is that `status` is sometimes a translated message, and sometimes "invalid", and when it's a message it's not probably user friendly enough for the initial setup

Reply:
Good point. I didn't see we have another case for `Token does not exist.`